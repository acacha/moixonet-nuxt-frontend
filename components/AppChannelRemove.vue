<template>
  <button @click="remove" data-test="remove-button">
    Eliminar
  </button>
</template>

<script>
export default {
  name: 'AppChannelRemove',
  props: {
    'channel': {
      type: Object,
      required: true
    }
  },
  methods: {
    remove () {
      this.$emit('removed', this.channel)
    }
  }
}
</script>
