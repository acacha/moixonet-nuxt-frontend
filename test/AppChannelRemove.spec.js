import { mount } from '@vue/test-utils'
import AppChannelRemove from '../components/AppChannelRemove'
import channelsFixture from '../cypress/fixtures/channels'

describe('AppChannelRemove', () => {
  let opts

  const build = (options) => {
    options = options || opts
    const wrapper = mount(AppChannelRemove, options)

    return {
      wrapper,
      removeButton: () => wrapper.find('[data-test=remove-button]')
    }
  }

  beforeEach(() => {
    opts = {
      propsData: {
        channel: channelsFixture[0]
      }
    }
  })

  test('is a Vue instance', () => {
    const { wrapper } = build()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders a list of channels', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('remove a channel when click on button', () => {
    const { wrapper, removeButton } = build()
    removeButton().trigger('click')
    expect(wrapper.emitted().removed).toBeTruthy()
    expect(wrapper.emitted().removed.length).toBe(1)
    expect(wrapper.emitted().removed[0][0]).toEqual(channelsFixture[0])
  })
})
