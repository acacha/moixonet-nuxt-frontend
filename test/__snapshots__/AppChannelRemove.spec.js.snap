// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppChannelRemove renders a list of channels 1`] = `
<button
  data-test="remove-button"
>
  
  Eliminar

</button>
`;
